test/types/community/collection/findX.test-d.ts

```diff
@@ -348,3 +348,21 @@ expectType<WithId<{ a: number; b: string }> | null>(
 expectType<WithId<{ a: number; b: string }> | null>(
   (await coll.findOneAndDelete({ a: 3 }, { projection: { _id: 0 } })).value
 );
+
+// NODE-3568: Uncomment when ModifyResult is removed in 5.0
+// expectType<Document> | null>((await coll.findOneAndDelete({ a: 3 })));
```

Comment:
For these expectations, the actual outcome will be `WithId<TSchema> | null` where `TSchema` is `{ a: number; b: string }` because of line 329

Reply:
Updated.